cloudinit/net/__init__.py

```diff
@@ -866,7 +866,11 @@ def get_interfaces():
         # skip nics that have no mac (00:00....)
         if name != 'lo' and mac == zero_mac[:len(mac)]:
             continue
-        ret.append((name, mac, device_driver(name), device_devid(name)))
+        # skip nics that have drivers blacklisted
+        driver = device_driver(name)
+        if blacklist_drivers is not None and driver in blacklist_drivers:
+            continue
```

Comment:
At the top of the function, 
```
if blacklist_drivers is None:
   blacklist_drivers = []
```
Then

```suggestion
        if driver in blacklist_drivers:
            continue
```